src/unity/toolkits/object_detection/object_detector.cpp

```diff
@@ -301,19 +301,33 @@ std::unique_ptr<model_spec> object_detector::init_model(
 }
 
 std::shared_ptr<MLModelWrapper> object_detector::export_to_coreml(
-    std::string filename) {
+    std::string filename, std::map<std::string, flexible_type> options) {
 
   // Initialize the result with the learned layers from the cnn_module.
   model_spec yolo_nn_spec(nn_spec_->get_coreml_spec());
 
   // Add the layers that convert to intelligible predictions.
-  add_yolo(&yolo_nn_spec, "coordinates", "confidence", "conv8_fwd",
+  std::string coordinates_str;
+  std::string confidence_str;
+  if (options["include_non_maximum_suppression"]=="True") {
+    coordinates_str = "raw_coordinates";
+    confidence_str = "raw_confidence";
+    //Set default values if thresholds not provided.
+    if (options.find("iou_threshold") == options.end())
+      options["iou_threshold"] = 0.45;
+    if (options.find("confidence_threshold") == options.end())
+      options["confidence_threshold"] = 0.25;
+  }
+  else {
+    coordinates_str = "coordinates";
+    confidence_str = "confidence";
```

Comment:
Just to make this code a bit more concise, can we initialize these variables assuming no NMS and then override in the if statement above? Then we don't need an else case at all.

Reply:
Good catch.